### ISSUE TITLE
Add non-core project "synapse" to serverless workflow organization

### ISSUE THREAD

**Issue by tsurdilo** [serverlessworkflow/specification#357]
Over the time we have built an eco-system around our specification which includes things like SDKs, Editor extensions, diagram generators etc. In order to grow it would be very beneficial to also include open-source runtime implementations which would like to be part of our project and hosted in our github organization.

@Neuroglia has kindly offered to donate his "Synapse" .NET 5.0 runtime implementation to our project.
This runtime can grow as part of our eco system.

I have prepared the https://github.com/serverlessworkflow/synapse

The project is intended to follow same rules as any other part of our specification eco system.

Our intention is as always to promote all runtimes that implement our specification, and also open the door for open-source runtimes to be part of our eco system. We are glad that we have the opportunity to start this with Synapse and open the door for more open-source runtimes that implement our specification to join if they wish.

2 maintainer's LGTM required (2/3)

- [x] Tihomir Surdilovic
- [ ] Manuel Stein
- [ ] Ricardo Zanini

**Comment by tsurdilo** [comment]
LGTM

**Comment by manuelstein** [comment]
> [...] an eco-system around our specification which includes things like SDKs, Editor extensions, diagram generators etc. [...]
> The project is intended to follow same rules as any other part of our specification eco system.

I like this proposal, because 
- it names Synapse a sub-project of the Serverless Workflow Specification, same as SDKs, editor extensions, diagram generator
- as such, it follows the CNCF model (CLA/signed commits, governance, code of conduct)
- reporting in Serverless Workflow community calls
- has its own release cycle, etc.

Hosting an implementation can be considered an endorsement or a promotion, so we should keep this in mind when linking implementations. It should not have a special status as compared to other runtime implementations.
Sorry I had suggested "reference implementation", on second thought that term has way too many connotations. So, I think sub-project is fine.

LGTM

**Comment by ricardozanini** [comment]
I agree with @manuelstein. I don't think "reference implementation" should be a good term, but since it's under the Serverless Workflow umbrella, people can understand as it is. We might need to reinforce that it is not.
